stan/math/prim/err/check_finite.hpp

```diff
@@ -1,144 +1,42 @@
- * @param name Variable name (for error messages)
- * @param y Variable to check
+ * @tparam T_y type of y
+ * @param function function name (for error messages)
+ * @param name variable name (for error messages)
+ * @param y variable to check
  * @throw <code>domain_error</code> if y is infinity, -infinity, or NaN
  */
-template <typename T_y, require_stan_scalar_t<T_y>* = nullptr>
+template <typename T_y, require_not_eigen_t<T_y>* = nullptr>
 inline void check_finite(const char* function, const char* name, const T_y& y) {
-  if (!internal::is_finite(y)) {
-    throw_domain_error(function, name, y, "is ", ", but must be finite!");
-  }
-}
-
-/**
- * Return <code>true</code> if all values in the std::vector are finite.
- *
- * @tparam T_y type of elements in the std::vector
- *
- * @param function name of function (for error messages)
- * @param name variable name (for error messages)
- * @param y std::vector to test
- * @return <code>true</code> if all values are finite
- **/
-template <typename T_y, require_stan_scalar_t<T_y>* = nullptr>
-inline void check_finite(const char* function, const char* name,
-                         const std::vector<T_y>& y) {
-  for (size_t n = 0; n < stan::math::size(y); n++) {
-    if (!internal::is_finite(stan::get(y, n))) {
-      throw_domain_error_vec(function, name, y, n, "is ",
-                             ", but must be finite!");
-    }
-  }
-}
-
-/**
- * Return <code>true</code> is the specified matrix is finite.
- *
- * @tparam Derived Eigen derived type
- *
- * @param function name of function (for error messages)
- * @param name variable name (for error messages)
- * @param y matrix to test
- * @return <code>true</code> if the matrix is finite
- **/
-template <typename EigMat, require_eigen_t<EigMat>* = nullptr>
-inline void check_finite(const char* function, const char* name,
-                         const EigMat& y) {
-  if (!value_of(y).allFinite()) {
-    for (int n = 0; n < y.size(); ++n) {
-      if (!std::isfinite(value_of_rec(y(n)))) {
-        throw_domain_error_vec(function, name, y, n, "is ",
-                               ", but must be finite!");
-      }
-    }
-  }
-}
-
-/**
- * Return <code>true</code> if all values in the std::vector are finite.
- *
- * @tparam T_y type of elements in the std::vector
- *
- * @param function name of function (for error messages)
- * @param name variable name (for error messages)
- * @param y std::vector to test
- * @return <code>true</code> if all values are finite
- **/
-template <typename T_y, require_not_stan_scalar_t<T_y>* = nullptr>
-inline void check_finite(const char* function, const char* name,
-                         const std::vector<T_y>& y) {
-  for (size_t n = 0; n < stan::math::size(y); n++) {
-    if (!internal::is_finite(stan::get(y, n))) {
-      throw_domain_error(function, name, "", "", "is not finite!");
-    }
-  }
+  auto is_good = [](const auto& y) { return std::isfinite(y); };
```

Comment:
Rather than a separate ```std::vector``` definition, you could use ```apply_vector_unary::apply()``` to use the ```allFinite``` member with both types in one definition

Reply:
I wasn't able to get this working.

THe input argument might be something like `std::vector<std::vector<var>>`, in which case the reduce still returns `std::vector<var>`.

Reply:
(Feel free to tell me how to do it though :P)

Reply:
Hmm, good point. A method for recursively applying Eigen checks to nested containers could be interesting, I'll have a think about it